packages/web-components/src/components/va-modal/va-modal.tsx

```diff
@@ -236,7 +236,11 @@ export class VaModal {
   }
 
   componentDidLoad() {
-    if (this.visible) this.setupModal();
+    if (this.visible) {
+      requestAnimationFrame
+        ? requestAnimationFrame(() => this.setupModal())
+        : setTimeout(() => this.setupModal(), 0);
```

Comment:
`requestAnimationFrame` queues `setupModal` right before the next repaint, ensuring the DOM is fully hydrated and ready for execution. The fallback to `setTimeout` ensures compatibility in older browsers where `requestAnimationFrame` is not available.